tgui/packages/tgui/interfaces/Modularshieldgen.tsx

```diff
@@ -0,0 +1,147 @@
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { Stack, Section, ProgressBar, Button, NumberInput } from '../components';
+
+type Modularshieldgendata = {
```

Comment:
I believe JS wants you to do camelCase for everything (snake_case is fine for stuff imported from DM) so this should be `modularShieldGeneratorData` instead.